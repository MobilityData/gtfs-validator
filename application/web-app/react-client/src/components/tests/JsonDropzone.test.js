/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React from "react";
import JsonDropzone from "../JsonDropzone";
import renderer from "react-test-renderer";

describe('JsonDropzone', () => {
    it('should be defined', () => {
        expect(JsonDropzone).toBeDefined();
    });

    it("should render correctly when on drag is not active", () => {
        let mockGetRootProps = jest.fn(() => {
            return jest.fn();
        });
        let mockGetInputProps = jest.fn(() => {
            return jest.fn();
        });
        let mockIsDragActive = false;
        let mockOnDrop = jest.fn();
        let mockUseDropZone = jest.fn(() => {
            return {
                getRootProps: mockGetRootProps,
                getInputProps: mockGetInputProps,
                isDragActive: mockIsDragActive
            };
        });
        const tree = renderer.create(
            <JsonDropzone id="id value" onDrop={mockOnDrop} accept="application/json" useDropzone={mockUseDropZone}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should render correctly when on drag is active", () => {
        let mockGetRootProps = jest.fn();
        let mockGetInputProps = jest.fn();
        let mockIsDragActive = true;
        let mockUseDropZone = jest.fn(() => {
            return {
                getRootProps: mockGetRootProps,
                getInputProps: mockGetInputProps,
                isDragActive: mockIsDragActive
            };
        });
        let mockOnDrop = jest.fn();
        const tree = renderer.create(
            <JsonDropzone id="id value" onDrop={mockOnDrop} accept="application/json" useDropzone={mockUseDropZone}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should call method when dropping element onto zone", () => {
        let mockOnDrop = jest.fn();
        let mockGetRootProps = jest.fn();
        let mockGetInputProps = jest.fn();
        let mockIsDragActive = false;
        let mockUseDropZone = jest.fn(() => {
            return {
                getRootProps: mockGetRootProps,
                getInputProps: mockGetInputProps,
                isDragActive: mockIsDragActive
            };
        });
        const tree = renderer.create(
            <JsonDropzone id="id value" onDrop={mockOnDrop} accept={"application/json"} useDropzone={mockUseDropZone}/>
        );
        tree.root.props.onDrop();
        expect(mockOnDrop).toHaveBeenCalledTimes(1);
    })
});
