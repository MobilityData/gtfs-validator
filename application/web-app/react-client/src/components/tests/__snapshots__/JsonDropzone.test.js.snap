// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonDropzone should render correctly when on drag is active 1`] = `
<div>
  <input
    className="dropzone-input"
    id="id value"
  />
  <div
    className="text-center"
  >
    <p
      className="dropzone-content"
    >
      Release to drop the files here
    </p>
  </div>
</div>
`;

exports[`JsonDropzone should render correctly when on drag is not active 1`] = `
<div
  _isMockFunction={true}
  getMockImplementation={[Function]}
  getMockName={[Function]}
  mock={
    Object {
      "calls": Array [],
      "instances": Array [],
      "invocationCallOrder": Array [],
      "results": Array [],
    }
  }
  mockClear={[Function]}
  mockImplementation={[Function]}
  mockImplementationOnce={[Function]}
  mockName={[Function]}
  mockRejectedValue={[Function]}
  mockRejectedValueOnce={[Function]}
  mockReset={[Function]}
  mockResolvedValue={[Function]}
  mockResolvedValueOnce={[Function]}
  mockRestore={[Function]}
  mockReturnThis={[Function]}
  mockReturnValue={[Function]}
  mockReturnValueOnce={[Function]}
>
  <input
    _isMockFunction={true}
    className="dropzone-input"
    getMockImplementation={[Function]}
    getMockName={[Function]}
    id="id value"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
  />
  <div
    className="text-center"
  >
    <p
      className="dropzone-content"
    >
      Drag 'n' drop your .json configuration file
    </p>
  </div>
</div>
`;
